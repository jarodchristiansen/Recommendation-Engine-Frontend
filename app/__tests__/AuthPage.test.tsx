// app/__tests__/AuthPage.test.tsx
import { render, act } from "@testing-library/react";
import Auth from "../auth/page";
import "@testing-library/jest-dom";
import { SessionProvider } from "next-auth/react";

describe("Auth Page", () => {
  it("matches the snapshot of the AuthPage", async () => {
    const { container } = await act(async () =>
      render(
        <SessionProvider>
          <Auth />
        </SessionProvider>
      )
    );

    expect(container).toMatchSnapshot();
  });
});
