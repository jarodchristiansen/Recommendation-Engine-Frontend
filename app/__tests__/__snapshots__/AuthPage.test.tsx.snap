// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth Page matches the snapshot of the AuthPage 1`] = `
<div>
  <div
    class="flex flex-col items-center justify-center h-screen bg-gradient-to-b from-gray-800 via-black to-gray-900 text-gray-100 p-6 gap-4"
  >
    <h1
      class="text-5xl font-bold mb-6 tracking-wider text-center"
    >
      Get in Tune with SpotRec
    </h1>
    <p
      class="text-lg text-gray-400 mb-8 text-center max-w-md"
    >
      Unlock personalized recommendations based on your favorite artists, albums, and listening habits.
    </p>
    <div>
      <div
        class="flex flex-col items center gap-6"
      >
        <p>
          Signed in as 
        </p>
        <button
          class="bg-red-500 text-white p-2 max-w-xs self-center font-bold rounded-md"
        >
          Sign out
        </button>
      </div>
    </div>
    <div
      class="mt-4 max-w-md text-center"
    >
      <p
        class="text-sm text-gray-500 font-bold"
      >
        No Spotify account? No worries! Signing in will guide you through a quick setup.
      </p>
    </div>
  </div>
</div>
`;
